keps/sig-apps/2214-indexed-job/README.md

```diff
@@ -148,19 +211,47 @@ because work lists can be implemented in a startup script using the completion
 index as building block.
 * The semantics of an indexed Job are similar to a StatefulSet, in the sense
 that Pods have an associated index.
-However, the APIs have a major difference: a StatefulSet doesn't have completion
-semantics, as opposed to Jobs.
+However, the APIs have major differences:
+  - a StatefulSet doesn't have completion semantics, as opposed to Jobs.
+  - a StatefulSet creates pods serially, whereas Job creates all Pods in
+    parallel.
+  - a StatefulSet gives Pods stable hostnames, a Job doesn't.
 
 [indexed Job]: https://github.com/kubernetes/community/blob/b21d1b27c8c748bf81283c2d89cde2becb5f2709/contributors/design-proposals/apps/indexed-job.md
 
 ### Risks and Mitigations
 
-Jobs have a known issue in which more than one Pod can be started even if
-parallelism and completion are set to 1 ([reference]). In the case of indexed
-Jobs, this translates to more than one Pod having the same index.
+- More than one pod created per index.
+
+  Jobs have a known rare issue in which more than one Pod can be started even if
+  parallelism and completion are set to 1 ([reference]). In the case of indexed
+  Jobs, this translates to more than one Pod having the same index.
+  
+  Just like for existing Job patterns, workloads have to handle duplicates at the
+  application level.
+
+- Scalability and latency of DNS programming, if users choose to pair the
+  Indexed Job with a headless service.
 
-Just like for existing Job patterns, workloads have to handle duplicates at the
-application level.
+  DNS programming requires the update of Endpoint or EndpointSlices by the
+  control plane and updating DNS records by the DNS provider.
+  This might not scale well for short-lived Jobs with high number of
+  parallelism.
+  
+  Moreoever, Pods need to be prepared to:
+  - Retry lookups in the case were the records didn't have time to update.
+  - The IP for a CNAME to change in the case of a Pod failure.
```

Comment:
this sentence doesn't read well and needs to start with a verb.

Reply:
updated